include/libsemigroups/sims.hpp

```diff
@@ -3120,16 +3120,16 @@ namespace libsemigroups {
    private:
     using node_type    = uint32_t;
     using KnuthBendix_ = KnuthBendix<word_type>;
-    std::vector<KnuthBendix_> _knuth_bendices;
-    Presentation<word_type>   _presentation;
+
+    std::thread::id                                   _default_thread_id;
```

Comment:
Do I understand correctly that `_default_thread_id` is just some not necessarily actually used thread id  value used so that we had some standard place to look for an already initialized Knuth-Bendix object? And this shouldn't cause issues if e.g. due to some coincidence `_default_thread_id` is in use, right?

Reply:
Yes that's correct, according to:

https://en.cppreference.com/w/cpp/thread/thread/id/id

> Default-constructs a new thread identifier. The identifier does not represent a thread.